packages/react/src/components/Forms/FileUploader/index.tsx

```diff
@@ -0,0 +1,347 @@
+      }
+      if (state === "uploading") {
+        return (
+          uploadingIcon ?? (
+            <Loader
+              size="small"
+              aria-label={t("components.forms.file_uploader.uploading")}
+            />
+          )
+        );
+      }
+      return icon ?? <span className="material-icons">upload</span>;
+    };
+
+    const renderCaption = () => {
+      if (state === "uploading") {
+        return t("components.forms.file_uploader.uploading");
+      }
+      if (bigText) {
+        return bigText;
+      }
+      return (
+        <>
+          {t("components.forms.file_uploader.caption")}
+          <span>{t("components.forms.file_uploader.browse_files")}</span>
+        </>
+      );
+    };
+
+    return (
+      <label
+        className={classNames(
+          "c__file-uploader",
+          "c__file-uploader--" + state,
+          {
+            "c__file-uploader--active": dragActive,
+            "c__file-uploader--animate-icon": animateIcon,
+          }
+        )}
+        onDragEnter={() => {
+          setDragActive(true);
+        }}
+        onDragLeave={(e) => {
+          /**
+           * This condition is important because onDragLeave is called when the cursor goes over
+           * a child of the current node, which is not intuitive. So here we need to make sure that
+           * the relatedTarget is not a child of the current node.
+           */
+          if (!container.current!.contains(e.relatedTarget as Node)) {
+            setDragActive(false);
+          }
+        }}
+        onDragOver={(e) => {
+          e.preventDefault();
+        }}
+        onDrop={(e) => {
+          // To prevent a new tab to open.
+          e.preventDefault();
+          const newFiles = Array.from(e.dataTransfer.files);
+          if (inputRef.current) {
+            inputRef.current.files = e.dataTransfer.files;
+            onFilesChange([...newFiles]);
+          }
+          setDragActive(false);
+        }}
+        ref={container}
+      >
+        <div className="c__file-uploader__inner">
+          {renderIcon()}
+          {children || (
+            <>
+              <div className="c__file-uploader__inner__caption">
+                {renderCaption()}
+              </div>
+              {text && (
+                <div className="c__file-uploader__inner__text">{text}</div>
+              )}
+            </>
+          )}
+          <input
+            type="file"
+            name={name}
+            ref={inputRef}
+            onChange={(e) => {
+              if (e.target.files) {
+                onFilesChange([...e.target.files]);
+              } else {
+                onFilesChange([]);
+              }
+            }}
+            multiple={multiple}
+            {...inputProps}
+          />
+        </div>
+      </label>
+    );
+  }
+);
+
+const FileUploaderMono = forwardRef<FileUploaderRefType, FileUploaderProps>(
```

Comment:
Same remark as other components, what do you think to split mono and multiple components ?

Reply:
let's do it